### PR TITLE
Teleporting objects use do_teleport rather than forcemove

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Replaces a bunch of teleporting things forceMove with do_teleport.

## Why It's Good For The Game

do_teleport respects no_teleport. Force move is for when something HAS to move there otherwise things break, not for teleports.

Anything that is used to teleport to a place with no_teleport = TRUE requires the forced flag.

## Changelog
:cl:
code: Replaces some teleporters forceMoves with do_teleports.
/:cl:

### CODE REVIEW

File: code/datums/dash_weapon.dm [hunk 1]
```diff
@@ -34,13 +34,15 @@
...
 	if(user in viewers(user.client.view, T))
 		var/obj/spot1 = new phaseout(get_turf(user), user.dir)
-		user.forceMove(T)
-		playsound(T, dash_sound, 25, 1)
-		var/obj/spot2 = new phasein(get_turf(user), user.dir)
-		spot1.Beam(spot2,beam_effect,time=20)
-		current_charges--
-		holder.update_action_buttons_icon()
-		addtimer(CALLBACK(src, .proc/charge), charge_rate)
+		if(do_teleport(user, T, channel = TELEPORT_CHANNEL_FREE), no_effects = TRUE)
```
Comment (PowerfulBacon) [inline code/datums/dash_weapon.dm]:
```suggestion
		if(do_teleport(user, T, channel = TELEPORT_CHANNEL_FREE, no_effects = TRUE))
```

File: code/modules/ruins/spaceruin_code/hilbertshotel.dm [hunk 2]
```diff
@@ -54,7 +54,7 @@ GLOBAL_VAR_INIT(hhmysteryRoomNumber, 1337)
         return
     if(ismob(loc))
         if(user == loc) //Not always the same as user
-            forceMove(get_turf(user))
+			forceMove(get_turf(user))
```
Comment (PowerfulBacon) [inline code/modules/ruins/spaceruin_code/hilbertshotel.dm]:
```suggestion
            forceMove(get_turf(user))
```

**Comment by ZombiesVsPlants** [comment]
does this mean bolted ai's can no longer be bluespace launchpadded?

**Comment by PowerfulBacon** [comment]
Bs launchpad already uses do teleport

**Review by utilbird — APPROVED** [review]

**Pull request merged**